source/adios2/ADIOSTypes.h

```diff
@@ -137,6 +114,35 @@ struct TypeInfo<T, typename std::enable_if<std::is_same<
     using ValueType = typename T::value_type;
 };
 
+const size_t UnknownDim = 0;
+const size_t JoinedDim = SIZE_MAX - 1;
```

Comment:
I remember numeric_limits wasn't well supported by C++11 compilers as constexpr a few years ago, so we end up using (size_t)-1 . If that has changed I think we should use numeric_limits<size_t> instead.